runner/src/internal.rs

```diff
@@ -103,21 +114,25 @@ impl std::fmt::Display for StatusResultValue {
 
 pub struct SingleStatusResult {
     value: StatusResultValue,
-    command: String,
     logs: String,
 }
 
-/// An execution step, which may be a single (dynamically dispatched) `Runnable`, or a collection of
+/// An execution step, which may be a single (dynamically dispatched) `Cmd`, or a collection of
 /// sub-steps.
 pub enum Step {
     Single {
         name: String,
-        runnable: Box<dyn Runnable>,
+        command: Cmd,
     },
     Multiple {
         name: String,
         steps: Vec<Step>,
     },
+    WithBackground {
+        name: String,
+        background: Cmd,
+        foreground: Box<Step>,
+    },
```

Comment:
To make sure I understand, 

when running examples, the "background" would be the server, and the foreground would be a client?

Reply:
Correct. The foreground may be multiple clients even, if we wanted to (but the background is a single Cmd, so that it may be stopped when done).

Reply:
`Box<Step>` is a single heap-allocated step, while `Vec<Step>` is a sequence of steps. The heap allocation is necessary because otherwise a self-referencing type without heap indirection would expand to an infinite type.

See https://doc.rust-lang.org/book/ch15-01-box.html#more-information-about-the-cons-list for a good overview.

Reply:
Oh I see, that makes. :) Thank you for sharing that link, was helpful to read.